secure/net/src/lib.rs

```diff
@@ -87,6 +87,7 @@ impl NetworkClient {
 
             let sleeptime = time::Duration::from_millis(100);
             while stream.is_err() {
+                debug!("Failed to connect to upstream {} {:?}", self.server, stream);
```

Comment:
Ahh, you'd like to print out the error?

I think you'd probably want to swap this to a loop:

```
loop {
  if let Some(stream) = stream {
    break;
  } else if let Err(e) = stream {
    debug!(..)
  }
  ...
}
```

You can also look into match statements.

Reply:
I think `while let` is what I'm going for, which is a `while` with a `match`.